### PR TITLE
Add tests for client heartbeat

### PR DESCRIPTION
**Pull request by tomalec**
TODO here or in a new PR:
 - tests for WS scenario,
 - tests for reconnect

### CODE REVIEW

File: test/specs/client/heartbeat.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+                        }), {name: 'reconnect'});
+                        
+                        // wait to send a ping
+                        await sleep(pingIntervalS*1*1000 + 1);
+                        // no response comes witihn pingIntervalS
+                        await sleep(pingIntervalS*1*1000 + 1);
+                        // do nothing for 1000ms
+                        await sleep(1000 + 1);
+                    });
+                    it('after 1000ms should call onReconnectionEnd', async () => {
```
Comment (warpech) [inline test/specs/client/heartbeat.js]: `after 1000ms ` redundantly repeated from `describe()`

**Comment by tomalec** [comment]
Addressed all, @warpech could you take another look?

**Comment by warpech** [comment]
LGTM!

**Comment by tomalec** [comment]
I'd prefer to merge this without waiting for the other, as I have branches that requires this one.

**Pull request merged**